File: Utils/LinkedLists/LinkedList.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

# TWO FUNCTION TO AID IN TERMINAL OUTPUT
def listToLinkedList(lst: list) -> ListNode:
    if not lst: return None

    head = ListNode(lst[0])
    current = head

    for val in lst[1:]:
        current.next = ListNode(val)
        current = current.next
    
    return head

def linkedListToList(head: ListNode) -> list:
    if not head: return

    res = []
    while head:
        res.append(head.val)
        head = head.next
    return res

File: Utils/TypeScript/LinkedList.ts
export class ListNode {
    val: number;
    next: ListNode | null;
    constructor(val?: number, next?: ListNode | null) {
        this.val = val === undefined ? 0 : val;
        this.next = next === undefined ? null : next;
    }
}

// TWO FUNCTIONS TO AID IN TERMINAL OUTPUT

export function listToLinkedList(list: any[]): ListNode | null {
    if (!list) return null;

    const head = new ListNode(list[0]);
    let current = head;

    for (let i = 1; i < list.length; i++) {
        current.next = new ListNode(list[i]);
        current = current.next;
    }

    return head;
}

export function linkedListToList(head: ListNode | null): any[] | undefined {
    if (!head) return;

    let current: ListNode | null = head;
    let res: any[] = [];

    while (current) {
        res.push(current.val);
        current = current.next;
    }

    return res;
}
